_data/meltano/extractors/tap-singer-jsonl/kgpayne.yml

```diff
@@ -0,0 +1,62 @@
+capabilities:
+- discover
```

Comment:
@kgpayne is this the full list of capabilities? What about catalog? state?

Reply:
@pnadolny13 as these are a thin wrapper on raw singer messages (rather than records), both tap and target are not SDK based and therefore don't get full functionality out-of-the-box 😅 Currently only discover is supported by the tap, though I do eventually want to support catalog (selection) and state. This effectively means it's up to the user to configure just the folders/s3 paths they want to read singer messages from, which has been good enough for my use-case.

In future it would be cool (though possibly not performant in the s3 case 🤔) to point at a top-level S3 'data lake' prefix with multiple streams (each in a folder) and allow selection purely via the catalog. I think state would work by bookmarking the last read file per stream (each file is timestamped), which should be relatively easy 👍

Reply:
Added [#10](https://github.com/kgpayne/tap-singer-jsonl/issues/10), [#11](https://github.com/kgpayne/tap-singer-jsonl/issues/11) and [#12](https://github.com/kgpayne/tap-singer-jsonl/issues/12) to track 🙂

Reply:
@pnadolny13 if you think its worth waiting for those to land, I'd be happy adding `hidden: true` to both of these (they operate best as a pair). I.e. `meltano run tap-ebay target-singer-jsonl tap-singer-jsonl target-csv` which is similar to `tap-ebay target-csv` but with either a local folder or S3 in between 🙂

Reply:
@kgpayne ah sorry yeah that makes total sense, forgot you werent using the SDK. It was odd not to see the usual capabilities. I dont think having those capabilities is a blocker from my perspective.